lib/gcloud/pubsub/topic.rb

```diff
@@ -299,85 +300,77 @@ def publish data = nil, attributes = {}
       #   default is `false`.
       #
-      # @return [Hash] Returns a hash that conforms to the following structure:
-      #
-      #   {
-      #     "etag"=>"CAE=",
-      #     "bindings" => [{
-      #       "role" => "roles/viewer",
-      #       "members" => ["serviceAccount:your-service-account"]
-      #     }]
-      #   }
+      # @return [Policy] the current Cloud IAM Policy for this topic
       #
       # @example Policy values are memoized to reduce the number of API calls:
       #   require "gcloud"
       #
       #   gcloud = Gcloud.new
       #   pubsub = gcloud.pubsub
-      #
       #   topic = pubsub.topic "my-topic"
-      #   puts topic.policy["bindings"]
-      #   puts topic.policy["rules"]
+      #
+      #   policy = topic.policy # API call
+      #   policy_2 = topic.policy # No API call
       #
       # @example Use `force` to retrieve the latest policy from the service:
       #   require "gcloud"
       #
       #   gcloud = Gcloud.new
       #   pubsub = gcloud.pubsub
-      #
       #   topic = pubsub.topic "my-topic"
-      #   policy = topic.policy force: true
+      #
+      #   policy = topic.policy force: true # API call
+      #   policy_2 = topic.policy force: true # API call
       #
       def policy force: nil
         @policy = nil if force
         @policy ||= begin
           ensure_service!
           grpc = service.get_topic_policy name
-          JSON.parse(Google::Iam::V1::Policy.encode_json(grpc))
+          Policy.from_grpc grpc
         rescue GRPC::BadStatus => e
           raise Error.from_error(e)
         end
       end
 
       ##
-      # Sets the access control policy.
+      # Updates the Cloud IAM access control policy for this topic. The policy
+      # should be read from {#policy}. See {Gcloud::Pubsub::Policy} for an
+      # explanation of the policy `etag` property and how to modify policies.
       #
-      # @param [String] new_policy A hash that conforms to the following
-      #   structure:
+      # @see https://cloud.google.com/pubsub/reference/rpc/google.iam.v1#iampolicy
+      #   google.iam.v1.IAMPolicy
       #
-      #     {
-      #       "bindings" => [{
-      #         "role" => "roles/viewer",
-      #         "members" => ["serviceAccount:your-service-account"]
-      #       }]
-      #     }
+      # @param [Policy] new_policy a new or modified Cloud IAM Policy for this
+      #   topic
       #
       # @example
       #   require "gcloud"
       #
       #   gcloud = Gcloud.new
       #   pubsub = gcloud.pubsub
-      #
-      #   viewer_policy = {
-      #     "bindings" => [{
-      #       "role" => "roles/viewer",
-      #       "members" => ["serviceAccount:your-service-account"]
-      #     }]
-      #   }
       #   topic = pubsub.topic "my-topic"
-      #   topic.policy = viewer_policy
+      #
+      #   policy = topic.policy # API call
+      #
+      #   policy.roles["roles/owner"] << "user:owner@example.com"
```

Comment:
I think it would be better to use a top-level policy method to make changes like this.

```ruby
policy.add "roles/owner", "user:owner@example.com"
```